aip/general/0193.md

```diff
@@ -34,272 +34,154 @@ is necessary, you **should** provide a link where a reader can get more
-  error.
-- For pre-existing (brownfield) APIs, the information in
-  `google.rpc.Status` already exists and can’t be changed. In this
-  case, do not edit anything in this `google.rpc.Status`, because some
-  users might have automation running against the pre-existing content.
-  For example, any changes in text for the `message` string could cause
-  breakages. For more information, see [Changing Error
-  Messages](#changing-error-messages).
-
-##### [google.rpc.Status payload][Status]
+### Status.code
 
-Details of this object are summarized in the following fields, field
-descriptions, and examples:
+The `code` field is the status code, which **must** be the numeric value of
+one of the elements of the [`google.rpc.Code`][Code] enum.
 
-- `code int32`
-  - The status code, which **must** be an enum value of
-    [`google.rpc.Code`][Code].
-  - *Example*: 8
-- `message string`
-  - A developer-facing error message, which **should** be in English.
-    Any user-facing error message **should** be located in the
-    LocalizedMessage payload.
-  - *Example message*: \"The zone \'us-east1-a\' does not have enough
-    resources available to fulfill the request. Try a different zone, or
-    try again later.\"
-- `details Object (repeated)`
-  - Additional text details about the error, which includes
-    `ErrorInfo`,`LocalizedMessage`, and so on.
-  - *Example*: Described in the [ErrorDetails](#error-details) section.
-
-
-**Important:** In the context of the `google.rpc.Status` protobuf
-message, the value of the field `code` is the numeric equivalent to the
-enum value chosen from [`google.rpc.Code`][Code]. For example, if
-[`INVALID_ARGUMENT`][InvalidArgument] is chosen, the value of `code`
-will be `3` (in the context of the Protobuf message,
-`google.rpc.Status`). However, when the error is expressed as JSON, as
-in the [sample above](#json-representation), the value of the field by
-the same name, `"code"`, is the HTTP status code equivalent of the
-selected `google.rpc.Code`. For example, the value of the field `"code"`
-in the JSON reply would be `400`.
-
-#### [ErrorDetails][details]
+For example, the value 5 is the numeric value of the `NOT_FOUND`
+enum element.
 
-Google defines a set of [standard detail payloads][details] for error
-details, which cover most common needs for API errors.  Services
-**should** use these standard detail payloads when feasible.
-
-Structured details with machine-readable identifiers **must** be used so
-that users can write code against specific aspects of the error. Error
-message strings **may** change over time; however, if an error message
-does not have a machine-readable identifier *in addition to* the `code`
-string, changing the error message **must** be considered a
-backwards-incompatible change.
+### Status.details
 
-The following section describes `ErrorInfo`, `LocalizedMessage`, and
-`Help` in more detail.
+The `details` field allows any multiple distinct messages with
```

Comment:
"any multiple distinct messages" reads a little awkwardly to me. What is this trying to say? "any number of distinct messages"?

Reply:
I agree it's awkward. It's basically trying to say "It's a repeated field, but you can't use the same message more than once". I think "message" makes all of this harder as well. (I haven't got a better alternative yet for that part though.)

Reply:
(The distinctness is covered later, so I've just removed it. See what you think.)